src/flags.rs

```diff
@@ -0,0 +1,192 @@
+fn test_set_flags_2() {
+  let (flags, rest) = set_flags(svec!["deno", "-r", "-D", "script.ts"]);
+  assert!(rest == svec!["deno", "script.ts"]);
+  assert!(
+    flags == DenoFlags {
+      help: false,
+      log_debug: true,
+      version: false,
+      reload: true,
+      allow_write: false,
+      allow_net: false,
+    }
+  );
+}
+
+#[test]
+fn test_set_flags_3() {
+  let (flags, rest) =
+    set_flags(svec!["deno", "-r", "script.ts", "-allow-write"]);
+  assert!(rest == svec!["deno", "script.ts"]);
+  assert!(
+    flags == DenoFlags {
+      help: false,
+      log_debug: false,
+      version: false,
+      reload: true,
+      allow_write: true,
+      allow_net: false,
+    }
+  );
+}
+
+// Returns args passed to V8, followed by args passed to JS
+// TODO Rename to v8_set_flags_preprocess
+fn parse_core_args(args: Vec<String>) -> (Vec<String>, Vec<String>) {
+  let mut rest = vec![];
+
+  // Filter out args that shouldn't be passed to V8
+  let mut args: Vec<String> = args
+    .into_iter()
+    .filter(|arg| {
+      if arg.as_str() == "--help" {
+        rest.push(arg.clone());
+        return false;
+      }
+
+      true
+    })
+    .collect();
+
+  // Replace args being sent to V8
+  for idx in 0..args.len() {
+    if args[idx] == "--v8-options" {
+      mem::swap(args.get_mut(idx).unwrap(), &mut String::from("--help"));
+    }
+  }
+
+  (args, rest)
+}
+
+#[test]
+fn test_parse_core_args_1() {
+  let js_args =
+    parse_core_args(vec!["deno".to_string(), "--v8-options".to_string()]);
+  assert!(js_args == (vec!["deno".to_string(), "--help".to_string()], vec![]));
+}
+
+#[test]
+fn test_parse_core_args_2() {
+  let js_args = parse_core_args(vec!["deno".to_string(), "--help".to_string()]);
+  assert!(js_args == (vec!["deno".to_string()], vec!["--help".to_string()]));
+}
+
+// Pass the command line arguments to v8.
+// Returns a vector of command line arguments that v8 did not understand.
+pub fn v8_set_flags(args: Vec<String>) -> Vec<String> {
+  // deno_set_flags(int* argc, char** argv) mutates argc and argv to remove
+  // flags that v8 understands.
+  // First parse core args, then converto to a vector of C strings.
+  let (argv, rest) = parse_core_args(args);
+  let mut argv = argv
+    .iter()
+    .map(|arg| CString::new(arg.as_str()).unwrap().into_bytes_with_nul())
+    .collect::<Vec<_>>();
+
+  // Make a new array, that can be modified by V8::SetFlagsFromCommandLine(),
+  // containing mutable raw pointers to the individual command line args.
+  let mut c_argv = argv
+    .iter_mut()
+    .map(|arg| arg.as_mut_ptr() as *mut i8)
+    .collect::<Vec<_>>();
+  // Store the length of the argv array in a local variable. We'll pass a
+  // pointer to this local variable to deno_set_flags(), which then
+  // updates its value.
+  let mut c_argc = c_argv.len() as c_int;
+  // Let v8 parse the arguments it recognizes and remove them from c_argv.
+  unsafe {
+    // TODO(ry) Rename deno_set_flags to deno_set_v8_flags().
```

Comment:
I'm a little surprised to see this as a TODO, as typing that comment was about as much work as actually making the change.